### PR TITLE
WIP to further improve cli handling

### PR DESCRIPTION
**Pull request by mr-tz**
Continuing to use the `-x` option to enable advanced options.

With these changes it's possible to use the advanced options without specifying `-x` all the time.

```
$ floss -h
usage: floss [-h] [-x] [-j] [-v] [-vv] [--color {auto,always,never}] [-d] [-q] [-s] sample

The FLARE team's open-source tool to extract obfuscated strings from malware.
  floss 2.0.0-rc1 - https://github.com/fireeye/flare-floss/

positional arguments:
  sample                path to sample to analyze

optional arguments:
  -h, --help            show this help message and exit
  -x                    enable eXpert arguments, see `floss --help -x`

rendering arguments:
  -j, --json            emit JSON instead of text
  -v, --verbose         enable verbose result document (no effect with --json)
  -vv, --vverbose       enable very verbose result document (no effect with --json)
  --color {auto,always,never}
                        enable ANSI color codes in results, default: only during interactive session

logging arguments:
  -d, --debug           enable debugging output on STDERR
  -q, --quiet           disable all status output except fatal errors

shellcode arguments:
  -s, --shellcode       analyze shellcode

only displaying core arguments, run `floss --help -x` to see all supported arguments

examples:
  extract all strings from an executable
    floss suspicious.exe

  extract all strings from shellcode
    floss -s shellcode.bin

```

```
$ floss -h -x
usage: floss [-h] [-x] [-j] [-v] [-vv] [--color {auto,always,never}] [-d] [-q] [-n MIN_LENGTH] [--functions [FUNCTIONS [FUNCTIONS ...]]] [--no-filter] [--max-instruction-count MAX_INSTRUCTION_COUNT]
             [--max-address-revisits MAX_ADDRESS_REVISITS] [--no-static-strings] [--no-decoded-strings] [--no-stack-strings] [-s] [--shellcode-entry-point SHELLCODE_ENTRY_POINT] [--shellcode-base SHELLCODE_BASE]
             [--shellcode-arch {i386,amd64}]
             sample

The FLARE team's open-source tool to extract obfuscated strings from malware.
  floss 2.0.0-rc1 - https://github.com/fireeye/flare-floss/

positional arguments:
  sample                path to sample to analyze

optional arguments:
  -h, --help            show this help message and exit
  -x                    enable eXpert arguments, see `floss --help -x`

rendering arguments:
  -j, --json            emit JSON instead of text
  -v, --verbose         enable verbose result document (no effect with --json)
  -vv, --vverbose       enable very verbose result document (no effect with --json)
  --color {auto,always,never}
                        enable ANSI color codes in results, default: only during interactive session

logging arguments:
  -d, --debug           enable debugging output on STDERR
  -q, --quiet           disable all status output except fatal errors

analysis arguments:
  -n MIN_LENGTH, --minimum-length MIN_LENGTH
                        minimum string length
  --functions [FUNCTIONS [FUNCTIONS ...]]
                        only analyze the specified functions, hex-encoded, like 0x401000
  --no-filter           do not filter deobfuscated strings (may result in many false positive strings)
  --max-instruction-count MAX_INSTRUCTION_COUNT
                        maximum number of instructions to emulate per function
  --max-address-revisits MAX_ADDRESS_REVISITS
                        maximum number of address revisits per function
  --no-static-strings   do not show static ASCII and UTF-16 strings
  --no-decoded-strings  do not show decoded strings
  --no-stack-strings    do not show stackstrings

shellcode arguments:
  -s, --shellcode       analyze shellcode
  --shellcode-entry-point SHELLCODE_ENTRY_POINT
                        shellcode entry point, hex-encoded like 0x401000
  --shellcode-base SHELLCODE_BASE
                        shellcode base offset, hex-encoded like 0x401000
  --shellcode-arch {i386,amd64}
                        shellcode architecture, default: autodetect

examples:
  only show strings of minimun length 6
    floss -n 6 suspicious.exe

  only show stack strings
    floss --no-static-strings --no-decoded-strings suspicious.exe

```

ref: #328

### CODE REVIEW

**Comment by williballenthin** [comment]
i like where this is going!

File: floss/main.py [hunk 1]
```diff
@@ -148,79 +180,95 @@ def make_parser(argv):
...
-            default=DEFAULT_MIN_LENGTH,
-            help="minimum string length",
-        )
+    analysis_group = parser.add_argument_group("analysis arguments")
+    analysis_group.add_argument(
+        "-n",
+        "--minimum-length",
+        dest="min_length",
+        default=DEFAULT_MIN_LENGTH,
+        help="minimum string length" if expert else argparse.SUPPRESS,
```
Comment (mr-tz) [inline floss/main.py:189]: `argparse.SUPPRESS` hides the help message in the "non-expert" mode, but the option can still be used

File: floss/main.py [hunk 2]
```diff
@@ -148,79 +180,95 @@ def make_parser(argv):
...
-            nargs="*",
-            help="only analyze the specified functions, hex-encoded, like 0x401000",
-        )
+    analysis_group.add_argument(
+        "--functions",
+        type=lambda x: int(x, 0x10),
+        nargs="+",
+        help="only analyze the specified functions, hex-encoded like 0x401000, space-separate multiple functions"
+        if expert
+        else argparse.SUPPRESS,
```
Comment (mr-tz) [inline floss/main.py:198]: `black`'s preferred way to format this :/

**Comment by mr-tz** [comment]
merging as cli and output are WIP anyway

**Pull request merged**